rules/no-empty-file.js

```diff
@@ -13,17 +13,25 @@ const isEmpty = node =>
 	|| node.type === 'EmptyStatement'
 	|| (node.type === 'ExpressionStatement' && 'directive' in node);
 
+const isTripleSlashDirective = node =>
+	node.type === 'Line' && node.value.startsWith('/');
+
+const hasTripeSlashDirectives = comments =>
+	comments.some(currentNode => isTripleSlashDirective(currentNode));
+
 /** @param {import('eslint').Rule.RuleContext} context */
 const create = context => {
 	const filename = context.getPhysicalFilename().toLowerCase();
+	const sourceCode = context.getSourceCode();
+	const comments = sourceCode.getAllComments();
 
 	if (!/\.(?:js|mjs|cjs|ts|mts|cts)$/.test(filename)) {
 		return {};
 	}
 
 	return {
 		Program(node) {
-			if (!isEmpty(node)) {
+			if (!isEmpty(node) || hasTripeSlashDirectives(comments)) {
```

Comment:
Can you check how `getAllComments()` works? Is it expensive to access? If true, we can inline that, so we don't need access it before `isEmpty()` check.

Reply:
I wrapped the `no-empty-file` test with `console.time()` and tried it as is and then with the looking for comments after the `isEmpty` check. There was no discernible difference in time. Granted, this is a super small use case.

Digging into ESLint, `getAllComments` is a [simple lookup](https://github.com/eslint/eslint/blob/dd58cd4afa6ced9016c091fc99a702c97a3e44f0/lib/source-code/source-code.js#L299-L301) on `ast.comments`, which [comes from the `SourceCode` constructor](https://github.com/eslint/eslint/blob/dd58cd4afa6ced9016c091fc99a702c97a3e44f0/lib/source-code/source-code.js#L176). 

It doesn't seem too overtly expensive, but more than happy to split it out after `isEmpty` to be safe.

Reply:
[Done](https://github.com/sindresorhus/eslint-plugin-unicorn/pull/1605/commits/d9f76cd8e51b3b90704290e4b608a5a77645738c)!